src/platform/forms-system/src/js/web-component-patterns/checkboxGroupPattern.jsx

```diff
@@ -0,0 +1,150 @@
+/* eslint-disable no-console */
+import { validateBooleanGroup } from '../validation';
+import VaCheckboxGroupField from '../web-component-fields/VaCheckboxGroupField';
+
+/**
+ * @param {WebComponentFieldProps} props
+ */
+const checkboxGroupItemUI = props => {
+  const { title, description, ...uiOptions } =
+    typeof props === 'object' ? props : { title: props };
+  return {
+    'ui:title': title,
+    'ui:description': description,
+    'ui:options': {
+      ...uiOptions,
+    },
+  };
+};
+
+/**
+ * Web component uiSchema for checkbox group
+ *
+ * Usage uiSchema:
+ * ```js
+ * checkboxGroup: checkboxGroupUI({
+ *    title: 'Checkbox group',
+ *    required: true,
+ *    labels: {
+ *      hasA: 'Option A',
+ *      hasB: 'Option B',
+ *    },
+ *  })
+ * ```
+ *
+ * ```js
+ * checkboxGroup: checkboxGroupUI({
+ *    title: 'Checkbox group',
+ *    hint: 'This is a hint',
+ *    required: false,
+ *    description: 'Please select at least one option',
+ *    labelHeaderLevel: '3',
+ *    tile: true,
+ *    labels: {
+ *      hasA: {
+ *        title: 'Option A',
+ *        description: 'Select this option if you want to do option A',
+ *      },
+ *      hasB: {
+ *        title: 'Option B',
+ *        description: 'Select this option if you want to do option B',
+ *      },
+ *    },
+ *    errorMessages: {
+ *       required: 'Please select at least one option',
+ *    },
+ *    validations: [validateBooleanGroup],
+ *  })
+ * ```
+ *
+ * Usage schema:
+ * ```js
+ * checkboxGroup: checkboxGroupSchema(['hasA', 'hasB'])
+ * ```
+ * @param {UIOptions & {
+ *  title?: UISchemaOptions['ui:title'],
+ *  required: boolean | (() => boolean),
+ *  labels: Record<string, string | UISchemaOptions>,
+ *  description?: UISchemaOptions['ui:description'],
+ *  tile?: boolean,
+ *  errorMessages?: UISchemaOptions['ui:errorMessages'],
+ *  labelHeaderLevel?: UISchemaOptions['ui:options']['labelHeaderLevel'],
+ *  hint?: string,
+ * }} options
+ * @returns {UISchemaOptions}
+ */
+export const checkboxGroupUI = ({
+  title,
+  description,
+  errorMessages,
+  labels,
+  required,
+  ...uiOptions
+}) => {
+  if (required === undefined) {
+    throw new Error(
```

Comment:
This is probably the most contentious part - welcome feedback on this - this requires you to specify the "required" property as part of using this pattern. I chose this approach because it could be easy to introduce a bug where you only specify the required part in the schema's `required: []`, which is normal procedure, but then forget to mark it here, which introduces the necessary validations in order for it to all work. the other option could be doing something like checkboxGroupRequiredUI, checkboxGroupOptionalUI.

Reply:
It's ok w/ me to always explicitly require.  :)